runtime/runtime/src/congestion_control.rs

```diff
@@ -234,24 +234,21 @@ impl ReceiptSinkV2 {
 
         let protocol_version = apply_state.current_protocol_version;
         let shard_layout = epoch_info_provider.shard_layout(&apply_state.epoch_id)?;
-        let shard_ids = if ProtocolFeature::SimpleNightshadeV4.enabled(protocol_version) {
-            shard_layout.shard_ids().collect_vec()
-        } else {
-            self.outgoing_limit.keys().copied().collect_vec()
-        };
-
-        let mut parent_shard_ids = BTreeSet::new();
-        for &shard_id in &shard_ids {
-            let parent_shard_id =
-                shard_layout.try_get_parent_shard_id(shard_id).map_err(Into::<EpochError>::into)?;
-            let Some(parent_shard_id) = parent_shard_id else {
-                continue;
+        let (shard_ids, parent_shard_ids) =
+            if ProtocolFeature::SimpleNightshadeV4.enabled(protocol_version) {
```

Comment:
Is it enough to rely on the protocol feature? Or should we wait until the first shard layout V2 is in effect? 

Just double checking that there isn't a delay between `SimpleNightshadeV4` enablement and the effective layout change.

Reply:
It should be fine as far as I can tell. 

The annoying consequence of this logic is that starting from this protocol feature only ShardLayoutV2 will be valid. I will try and make it a bit safer and forced by adding some good assertions later.

Reply:
Actually ShardLayoutV1 with and empty split map should also be ok so it's not that bad. Still I'll make it nicer when I get the time.